### DISCUSSION TITLE
How to persist server side cache with client side data fetching while using next-urql package.

### DISCUSSION THREAD

**Discussion by xoldd** [urql-graphql/urql#2618] [Q&A] [answered]
I have a list of items. I want to initially server side render some of those items. And the rest I want to fetch on the client side using pagination when the client asks for it. How to do this with next-urql package? I tried doing it but the server side cache seems to be wiped out if I enclose the exported nextjs page inside `withClient` function from next-urql. It makes a new request to graphql server for the items I wanted to be server side rendered. How am I supposed to use urql on the client side with this approach?

![Screenshot from 2022-08-18 00-21-55](https://user-images.githubusercontent.com/82873939/185219985-6b8ced80-eac4-4fda-ba9c-3f253be4e65b.png)
![Screenshot from 2022-08-18 00-21-41](https://user-images.githubusercontent.com/82873939/185219991-1d3088cd-afa1-4961-86a7-c804417306a4.png)

Also I'm trying to make use of urql's graphcache package for cache exchange. I want the server side cache to show up in the urql devtools.

**Comment by kitten** [comment, marked as answer]
Hiya 👋

In the first image you sent, pay close attention to the `exchanges` array you've got there. In there you have the devtools exchange and all default exchanges, however, that `ssrExchange` argument is where the magic happens.

That's basically a tiny cache that's only responsible for handing over SSR results to the client-side. If you compare your code to snippets on this docs page, you'll see that it basically stands in for the `fetchExchange` and "repeats" results that the SSR server has already provided: https://formidable.com/open-source/urql/docs/advanced/server-side-rendering/

In short, you have to include it in the list of exchanges at that exact position

**Reply by xoldd** [reply]
![Screenshot from 2022-08-18 13-45-26](https://user-images.githubusercontent.com/82873939/185345551-7b30de6d-f73f-45ae-9b74-593fa48dafcc.png)
Thanks. Added ssrExchange in the exchanges array before fetchExchange and it started to work. Let me know if I did it correctly.

**Reply by xoldd** [reply]
![Screenshot from 2022-08-18 16-35-39](https://user-images.githubusercontent.com/82873939/185380626-8e25c9d7-33c4-4422-80de-36391893eb5b.png)
@kitten In this function where do we pass the variables used in the query defintion. Like this one :-
![Screenshot from 2022-08-18 16-38-32](https://user-images.githubusercontent.com/82873939/185380992-1500bb34-5ee5-41b5-9037-65d387b9a155.png)

**Reply by JoviDeCroock** [reply]
@xoldyckk https://formidable.com/open-source/urql/docs/api/core/#clientquery

**Reply by xoldd** [reply]
I'm so dumb. I thought query and useQuery would have different options. Thanks.
